packages/cashc/src/semantic/TypeCheckTraversal.ts

```diff
@@ -51,7 +52,21 @@ export default class TypeCheckTraversal extends AstTraversal {
 
   visitTupleAssignment(node: TupleAssignmentNode): Node {
     node.tuple = this.visit(node.tuple);
-    expectAssignable(node.createVariableDefNode('name'), (node.tuple.type as TupleType).elementType, node.type);
+    if (!(node.tuple instanceof BinaryOpNode) || node.tuple.operator !== BinaryOperator.SPLIT) {
+      throw new TupleAssignmentError(node.tuple);
+    }
+    const tupleType = node.tuple.left.type;
+    for (const variable of [node.var1, node.var2]) {
+      if (!implicitlyCastable(tupleType, variable.type)) {
+        // Ignore if both are of type byte. problem: bytes16 can be typed to bytes32
```

Comment:
Right that is a bit of a difficult situation. We might be able to infer some types, like if we put in a `bytes32` and say `split(10)` then we know that the types are `bytes10` and `bytes22`. But that can get complicated, especially if the second parameter isn't a literal (e.g. `x.split(y)`), so I think we shouldn't implement that here. Perhaps something for later.

Reply:
If you have any other ideas about this I'd love to hear it, but I guess we might just have to leave it like this.